analyzers/tests/SonarAnalyzer.Test/Trackers/ArgumentTrackerTest.cs

```diff
@@ -0,0 +1,1290 @@
+            context.Parameter.ContainingType.Name.Should().Be("ProcessStartInfo");
+        }
+    }
+
+    [DataTestMethod]
+    [DataRow("""ProcessStartInfo($$"fileName")""", "fileName", 0, true)]
+    [DataRow("""ProcessStartInfo($$"fileName")""", "arguments", 1, false)]
+    [DataRow("""ProcessStartInfo("fileName", $$"arguments")""", "arguments", 1, true)]
+    [DataRow("""ProcessStartInfo("fileName", $$"arguments")""", "arguments", 0, false)]
+    [DataRow("""ProcessStartInfo($$"fileName", "arguments")""", "arguments", 1, false)]
+    [DataRow("""ProcessStartInfo(arguments:= $$"arguments", fileName:= "fileName")""", "arguments", 1, true)]
+    [DataRow("""ProcessStartInfo(arguments:= $$"arguments", fileName:= "fileName")""", "fileName", 0, false)]
+    [DataRow("""ProcessStartInfo(arguments:= "arguments", $$fileName:= "fileName")""", "fileName", 0, true)]
+    [DataRow("""ProcessStartInfo(arguments:= "arguments", $$fileName:= "fileName")""", "arguments" +
+        "", 1, false)]
+    public void Constructor_SimpleArgument_VB(string constructor, string parameterName, int descriptorPosition, bool expected)
+    {
+        var snippet = $$"""
+            Dim a = New System.Diagnostics.{{constructor}}
+            """;
+        var context = ArgumentContextVB(WrapInMethodVB(snippet));
+
+        var descriptor = ArgumentDescriptor.ConstructorInvocation(KnownType.System_Diagnostics_ProcessStartInfo, parameterName, descriptorPosition);
+        var result = MatchArgumentVB(context, descriptor);
+        result.Should().Be(expected);
+    }
+
+    [DataTestMethod]
+    [DataRow("""($$"fileName")""", "fileName", 0, true)]
+    [DataRow("""($$"fileName")""", "arguments", 1, false)]
+    [DataRow("""("fileName", $$"arguments")""", "arguments", 1, true)]
+    [DataRow("""("fileName", $$"arguments")""", "arguments", 0, false)]
+    [DataRow("""($$"fileName", "arguments")""", "arguments", 1, false)]
+    [DataRow("""(arguments: $$"arguments", fileName: "fileName")""", "arguments", 1, true)]
+    [DataRow("""(arguments: $$"arguments", fileName: "fileName")""", "fileName", 0, false)]
+    [DataRow("""(arguments: "arguments", $$fileName: "fileName")""", "fileName", 0, true)]
+    [DataRow("""(arguments: "arguments", $$fileName: "fileName")""", "arguments", 1, false)]
+    public void Constructor_TargetTyped(string constructor, string parameterName, int descriptorPosition, bool expected)
+    {
+        var snippet = $$"""
+            System.Diagnostics.ProcessStartInfo psi = new{{constructor}};
+            """;
+        var context = ArgumentContextCS(WrapInMethodCS(snippet));
+
+        var descriptor = ArgumentDescriptor.ConstructorInvocation(KnownType.System_Diagnostics_ProcessStartInfo, parameterName, descriptorPosition);
+        var result = MatchArgumentCS(context, descriptor);
+        result.Should().Be(expected);
+    }
+
+    [DataTestMethod]
+    [DataRow("""new Dictionary<TKey, TValue>($$1)""", "capacity", 0, true)]
```

Comment:
All the cases are returning `true`. You can either remove the parameter, or add some cases where there are no matches.

e.g.
```csharp
    [DataRow("""new Dictionary<TKey, TValue>($$1)""", "wrongParamName", 0, false)]
```